src/app.rs

```diff
@@ -156,46 +156,59 @@ impl App {
             None => None,
         }
     }
-
     pub fn next_program(&mut self) {
         let items = self.items.lock().unwrap();
         let mut state = self.state.lock().unwrap();
-
-        let i = match state.selected() {
-            Some(i) => {
-                if i >= items.len() - 1 {
-                    0
-                } else {
-                    i + 1
+        if items.len() > 0 {
```

Comment:
nit: could you move acquiring the `state` lock to inside this if? Same for `prev_program`. No need to get the lock if we know we are not gonna modify the state.

Reply:
done